tests/api/v1/collectors/heartbeat.js

```diff
@@ -171,22 +168,6 @@ describe('tests/api/v1/collectors/heartbeat.js >', () => {
         .end(done);
       });
 
-      it('valid collector token, doesnt match collector - by name', (done) => {
-        api.post(`/v1/collectors/${collector1.name}/heartbeat`)
```

Comment:
Why are you removing this?

Reply:
This was removed to match the deleted lines from 238 to 243 in the heartbeat code.

Reply:
This is a legitimate test, right? If deleting those lines causes this test to fail, then they shouldn't be deleted.
There was a TODO in there that they should be removed when it was working in the middleware. If it's not, then they shouldn't be removed yet.

Reply:
This is not a legitimate test. There is a story to make sure that this does not happen.

Reply:
Are you saying it's ok to send a heartbeat for collector A with a token registered to collector B? If that's not allowed then we need this test. If there's a story to move it to the middleware, it's still going to need to be tested.

I don't think we should move it until it's ready, and we shouldn't delete the test now only to add it back later.

Reply:
It is not ok for a collector A to send a heartbeat with a token registered to collector B. Lines 238 to 243 and these tests should not have been checked in. I missed it while I was reviewing the PR. There is a story to handle it, the code and its related tests should be added as a part of the story. We should be adding code now to be deleted later.